RecoLocalCalo/HGCalRecAlgos/interface/HGCalMultiCluster.h

```diff
@@ -0,0 +1,93 @@
+#ifndef RecoLocalCalo_HGCalRecAlgos_HGCalMultiCluster
+#define RecoLocalCalo_HGCalRecAlgos_HGCalMultiCluster
+
+#include <vector>
+
+#include "DataFormats/EgammaReco/interface/BasicCluster.h"
+
+
+class HGCalMultiCluster{
+
+public:
+  typedef std::vector<reco::BasicCluster>::const_iterator component_iterator;
+  typedef std::vector<reco::BasicCluster> ClusterCollection;
+
+  HGCalMultiCluster(){
+  }
+  HGCalMultiCluster(ClusterCollection &thecls) : myclusters(thecls){
+  }
+  HGCalMultiCluster(const reco::BasicCluster &thecl) :  myclusters(1,thecl){
+  }
+  void push_back(const reco::BasicCluster &b){
+    myclusters.push_back(b);
+  }
+  unsigned int size()
+  {
+    return myclusters.size();
+  }						
+  component_iterator begin() const {
+    return myclusters.begin();
+  }
+  component_iterator end() const {
+    return myclusters.end();
+  }
+  double simple_z(double vz) const {
+    double acc = 0.0;
+    double totweight = 0.;
+    for(component_iterator it = begin(); it != end(); it++){
+      acc += (it->z()-vz)*it->energy()*it->size();
+      totweight+=it->energy()*it->size();
+    }
+    return acc/totweight;
```

Comment:
@lgray: The public interface of this class provides a constructor that takes no arguments and allows `myclusters` to be of zero size. That certainly suggests that `simple_z` could be called with the null set. In the constructor that takes a `ClusterCollection`, it would be good to add `assert(size() > 0);` to ensure that the input collection isn't empty.